test/tests/functional/pbs_cray_vnode_pool.py

```diff
@@ -0,0 +1,337 @@
+                                attrib=attr_A, expect=False)
+        except PbsManagerError as e:
+            self.assertEquals(
+                e.msg[1], "qmgr: Error (15014) returned from server")
+
+        attr_A = {'vnode_pool': '0'}
+        try:
+            self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA,
+                                attrib=attr_A, expect=False)
+        except PbsManagerError as e:
+            self.assertEquals(
+                e.msg[1], "qmgr: Error (15014) returned from server")
+
+        attr_A = {'vnode_pool': 'a'}
+        try:
+            self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA,
+                                attrib=attr_A, expect=False)
+        except PbsManagerError as e:
+            self.assertEquals(
+                e.msg[1], "qmgr: Error (15014) returned from server")
+
+    def test_two_moms_single_vnode_pool(self):
+        """
+        Same vnode_pool for two moms shall result in one mom being the
+        inventory mom and the other the non-inventory mom.
+        The inventory mom goes down (e.g. killed).
+        Compute nodes remain up even when the inventory mom is killed,
+        since another mom is reporting them. Bring up killed mom.
+        Check that a new inventory mom is listed in the log.
+        """
+        self.server.manager(MGR_CMD_SET, SERVER, {"log_events": -1})
+
+        found_in_momA = 0
+        found_in_momB = 0
+        self.momA = self.moms.values()[0]
+        try:
+            self.momB = self.moms.values()[1]
+        except IndexError:
+            self.assertTrue(
+                False, msg="Provide at least 2 moms while invoking test")
+
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+
+        attr = {'vnode_pool': '1'}
+
+        start_time = int(time.time())
+
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA, attrib=attr)
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostB, attrib=attr)
+
+        rc = self.server.log_match("Mom %s added to vnode_pool %s" %
+                                   (self.hostB, '1'), max_attempts=5,
+                                   starttime=start_time)
+        self.assertTrue(rc)
+
+        rc = self.momA.log_match(
+            "Hello (no inventory required) from server", max_attempts=9,
+            starttime=start_time)
+        if rc is None:
+            found_in_momA = 0
+        else:
+            found_in_momA = 1
+
+        rc = self.momB.log_match(
+            "Hello (no inventory required) from server", max_attempts=9,
+            starttime=start_time)
+        if rc is None:
+            found_in_momB = 0
+        else:
+            found_in_momB = 1
+
+        self.assertEqual(found_in_momA + found_in_momB,
+                         1, msg="an inventory mom not chosen correctly")
+
+        # Only one mom is inventory mom
+        if (found_in_momA == 0):
```

Comment:
This looks fine to me. At this point either "found_in_momA" or "found_in_momB" could have a value of 0 because we are not guaranteed which mom will be the inventory mom.  After line 154 we know there is only one inventory mom.  If "found_in_momA" is set to 0 then it did not get hello_no_inventory which makes momA the inventory mom.  If "found_in_momA" is not set to 1 then it got hello_no_inventory which makes momB the inventory mom.